File: php/src/Resource/Decisions.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Resource;

use Fivexer\SDK\Fivexer;
use Fivexer\SDK\Model\Decision;

/**
 * Decision-trace resource (explainability). Delegates to Fivexer::request().
 */
final class Decisions
{
    public function __construct(private readonly Fivexer $client)
    {
    }

    /**
     * Query decision traces: who was matched, and why.
     *
     * @param string|null $taskId Filter to a single task
     * @param string|null $workerId Filter to a single worker
     * @param int<1, max>|null $limit Max traces to return
     * @return list<Decision> Newest first
     */
    public function list(?string $taskId = null, ?string $workerId = null, ?int $limit = null): array
    {
        $data = $this->client->request('GET', '/decisions', null, [
            'taskId' => $taskId,
            'workerId' => $workerId,
            'limit' => $limit,
        ]) ?? [];
        return \array_map(
            static fn(array $d): Decision => Decision::fromArray($d),
            $data['decisions'] ?? [],
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/Decisions.java
package io.fivexer.sdk;

import io.fivexer.sdk.model.Decision;
import io.fivexer.sdk.model.DecisionList;
import java.util.Collections;
import java.util.List;

/** Match decisions: who was considered for a task, and why the winner won. */
public final class Decisions {

    private final Fivexer client;

    Decisions(Fivexer client) {
        this.client = client;
    }

    public List<Decision> list(ListDecisionsQuery query) {
        ListDecisionsQuery q = query == null ? new ListDecisionsQuery() : query;
        DecisionList envelope = client.request("GET", "/decisions", null, q.toQuery(), DecisionList.class, false);
        return envelope.getDecisions() == null ? Collections.emptyList() : envelope.getDecisions();
    }

    public List<Decision> list() {
        return list(null);
    }
}
